components/style/values/specified/grid.rs

```diff
@@ -301,3 +340,377 @@ impl<L: ToComputedValue> ToComputedValue for TrackSize<L> {
         }
     }
 }
+
+/// Parse the grid line names into a vector of owned strings.
+///
+/// https://drafts.csswg.org/css-grid/#typedef-line-names
+pub fn parse_line_names(input: &mut Parser) -> Result<Vec<String>, ()> {
+    input.expect_square_bracket_block()?;
+    input.parse_nested_block(|input| {
+        let mut values = vec![];
+        loop {
+            let ident = input.expect_ident()?;
+            if ident.eq_ignore_ascii_case("span") {
+                return Err(())
+            }
+
+            values.push(ident.into_owned());
+            if input.try(|i| i.expect_whitespace()).is_err() {
+                break
+            }
+        }
+
+        Ok(values)
+    })
+}
+
+/// The initial argument of the `repeat` function.
+///
+/// https://drafts.csswg.org/css-grid/#typedef-track-repeat
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum RepeatCount {
+    /// A positive integer. This is allowed only for `<track-repeat>` and `<fixed-repeat>`
+    Number(u32),
```

Comment:
Oh, that's just to be consistent with a CSS integer (since it's an index, it doesn't have to be signed, and `usize` is platform-specific, so something definitive).